### PR TITLE
Email Verification Window

### PR DESCRIPTION
**Pull request by DRvanR**
https://www.pivotaltracker.com/story/show/86193476

### CODE REVIEW

File: app/AppKernel.php [hunk 1]
```diff
@@ -36,7 +36,7 @@ public function registerBundles()
             new Doctrine\Bundle\MigrationsBundle\DoctrineMigrationsBundle(),
             new Surfnet\StepupMiddleware\CommandHandlingBundle\SurfnetStepupMiddlewareCommandHandlingBundle(),
             new Surfnet\StepupMiddleware\ApiBundle\SurfnetStepupMiddlewareApiBundle(),
-            new Surfnet\StepupMiddleware\MiddlewareBundle\MiddlewareBundle(),
+            new Surfnet\StepupMiddleware\MiddlewareBundle\SurfnetStepupMiddlewareMiddlewareBundle(),
```
Comment (rjkip) [inline app/AppKernel.php]: :sweat_smile:

File: src/Surfnet/Stepup/Tests/DateTime/DateTimeTest.php [hunk 2]
```diff
@@ -0,0 +1,119 @@
...
+    /**
+     * Ensure that the __toString of our DateTime object actually uses the correct format. For the reason why, read the
+     * docblock above the {@see the_configured_format_is_what_is_needed_for_correct_application_behavior()} test
+     *
+     * @test
+     * @group domain
+     */
+    public function to_string_returns_the_time_in_the_correct_format()
+    {
+        $coreDateTimeObject = new CoreDateTime('@1000');
```
Comment (rjkip) [inline src/Surfnet/Stepup/Tests/DateTime/DateTimeTest.php]: Tests, yay!

**Comment by DRvanR** [comment]
woohoo

@rjkip

File: src/Surfnet/Stepup/Identity/Entity/UnverifiedSecondFactor.php [hunk 3]
```diff
@@ -35,6 +35,7 @@
  * address to verify this second factor.
  *
  * @SuppressWarnings(PHPMD.CouplingBetweenObjects)
+ * @SuppressWarnings(PHPMD.UnusedPrivateFields)
```
Comment (rjkip) [inline src/Surfnet/Stepup/Identity/Entity/UnverifiedSecondFactor.php]: Eh?

File: src/Surfnet/Stepup/Identity/Identity.php [hunk 4]
```diff
@@ -51,6 +52,7 @@
 /**
  * @SuppressWarnings(PHPMD.CouplingBetweenObjects)
  * @SuppressWarnings(PHPMD.TooManyMethods)
+ * @SuppressWarnings(PHPMD.ExcessiveClassComplexity)
```
Comment (rjkip) [inline src/Surfnet/Stepup/Identity/Identity.php]: Hory crapu, never seen that one before :sweat_smile:

**Comment by rjkip** [comment]
Bar the `UnusedPrivateFields` (why?), :+1:

**Comment by DRvanR** [comment]
W.r.t. the `UnusedPrivateField`, this is due to the usage of the (now superseded) `emailVerificationRequestedAt`. Removing this will be a BC break, will discuss if this is an issue since we're still in development...

**Comment by DRvanR** [comment]
@rjkip Removed the `UnusedPrivateField` suppression as discussed, could you please check?

**Comment by rjkip** [comment]
As far as I can see :+1:

**Pull request merged**